R/sdf_interface.R

```diff
@@ -694,3 +694,36 @@ sdf_broadcast <- function(x) {
                 "broadcast", sdf) %>%
     sdf_register()
 }
+
+#' Repartition a Spark DataFrame
+#'
+#' @template roxlate-ml-x
+#'
+#' @param partitions number of partitions
+#' @param columns vector of column names used for partitioning, only supported for Spark 2.0+
+#'
+#' @export
+sdf_repartition <- function(x, partitions = NULL, columns = NULL) {
+  sdf <- spark_dataframe(x)
+  sc <- spark_connection(sdf)
+
+  partitions <- partitions %||% 0L
```

Comment:
Maybe I'm misunderstanding something, but it looks like we don't allow `NULL` here (we remap that to `0L`) -- so is the `allow.null` below needed?